File: backend/app/schemas/company.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class CompanyCreate(BaseModel):
    name: str
    legal_name: str | None = None
    website_url: str | None = None
    description: str | None = None
    headquarters: str | None = None
    jurisdictions: list[str] = []


class CompanyUpdate(BaseModel):
    name: str | None = None
    legal_name: str | None = None
    website_url: str | None = None
    description: str | None = None
    headquarters: str | None = None
    jurisdictions: list[str] | None = None


class CompanyRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    legal_name: str | None
    website_url: str | None
    description: str | None
    headquarters: str | None
    jurisdictions: list[str]
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/company.ts
export interface Company {
  id: string;
  name: string;
  legal_name: string | null;
  website_url: string | null;
  description: string | null;
  headquarters: string | null;
  jurisdictions: string[];
  created_at: string;
  updated_at: string;
}

export interface CompanyCreate {
  name: string;
  legal_name?: string;
  website_url?: string;
  description?: string;
  headquarters?: string;
  jurisdictions?: string[];
}
